### ISSUE TITLE
`prune`: filtering out `ADD --checksum=... https://...` entries

### ISSUE THREAD

**Issue by fenollp** [docker/buildx#2448]
### Description

I'd like to be able to run `docker buildx prune` but keep in cache the *pinned* `ADD` downloads.
Could we extend on the `--filter=..` syntax?

Side curiosity: why are these cache entries marked `Mutable: true`?
From my understanding of [`du`'s docs](https://github.com/docker/buildx/blob/d4f088e689b41353d74f1a0bfcd6d7c0b213aed2/docs/reference/buildx_du.md?plain=1#L55C15-L57C56) that'd mean that its size can grow.

Is there a missed optimization opportunity of in the `ADD` op here, setting `mutable=false`?

**Comment by tonistiigi** [comment]
We could change the type for these to `source.http` like `source.git.checkout` and `source.local` instead of `regular`.

Mutable is because there was no need to make it immutable (there are no layers on top of it or images created directly from this data). It doesn't change anything for the cache reuse. Actual changing of the data is theoretical, currently there isn't any option that would for example add new files to same snapshot, or continue download or reuse previous snapshot of downloaded file and for example rename the file.

**Comment by tonistiigi** [comment]
Yes, you can use `docker buildx prune --filter 'type=source.local'` for example but looks like the records created from `ADD url` are type=regular atm and don't have a specific type definition.
